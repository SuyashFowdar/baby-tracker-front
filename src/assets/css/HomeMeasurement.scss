.measure {
  height: calc((100vh - 144px) / 2);
  width: calc(50% - 20px);
  margin: 10px;
  overflow: hidden;
  border: 1px solid rgb(180, 180, 180);
  border-radius: 10px;
  background-color: rgb(255, 255, 255);

  .amount {
    padding: 10px;
    font-size: 36px;

    &:not(.new) {
      color: rgb(0, 150, 255);
    }
  }

  .dif {
    font-size: 14px;
    margin-bottom: 10px;
    margin-top: -10px;
    font-weight: bold;
  }
}
