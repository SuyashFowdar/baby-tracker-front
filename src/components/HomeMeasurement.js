import React from 'react';
import PropTypes from 'prop-types';
import { HashRouter as Router, Link } from 'react-router-dom';
import '../assets/css/HomeMeasurement.scss';

const HomeMeasurement = ({ measure }) => {
  const { measurements } = measure;

  return (
    <Router>
      <Link to={`/measure/${measure.id}`} className="measure link bg-cover col cross-center main-center">
        {measurements && measurements.length
          ? (
            <div>
              <div className="amount">{`${measurements[measurements.length - 1].amount} ${measure.unit}`}</div>
              {measurements[measurements.length - 2]
                ? (
                  <div className="row main-center dif" style={{ color: measurements[measurements.length - 1].amount > measurements[measurements.length - 2].amount ? 'green' : 'red' }}>
                    {measurements[measurements.length - 1].amount
                    > measurements[measurements.length - 2].amount ? '+' : ''}
                    {(measurements[measurements.length - 1].amount
                    - measurements[measurements.length - 2].amount).toFixed(2)}
                  </div>
                )
                : <div />}
            </div>
          )
          : <div className="amount new">Add new</div>}
        <div className="type">{measure.item}</div>
      </Link>
    </Router>
  );
};

HomeMeasurement.defaultProps = {
  measure: {},
};

HomeMeasurement.propTypes = {
  measure: PropTypes.instanceOf(Object),
};

export default HomeMeasurement;
