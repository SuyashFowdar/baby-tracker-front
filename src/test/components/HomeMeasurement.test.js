import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import HomeMeasurement from '../../components/HomeMeasurement';
import reducer from '../../reducers/index';

it('renders correctly', () => {
  const store = createStore(reducer);
  const tree = renderer
    .create((
      <Provider store={store}>
        <HomeMeasurement />
      </Provider>
    ))
    .toJSON();
  expect(tree).toMatchSnapshot();
});
