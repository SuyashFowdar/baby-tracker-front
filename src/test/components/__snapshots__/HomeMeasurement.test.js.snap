// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  className="measure link bg-cover col cross-center main-center"
  href="#/measure/undefined"
  onClick={[Function]}
>
  <div
    className="amount new"
  >
    Add new
  </div>
  <div
    className="type"
  />
</a>
`;
